pulser-core/pulser/channels.py

```diff
@@ -94,10 +94,14 @@ class Channel(ABC):
     mod_bandwidth: Optional[float] = None  # MHz
 
     @property
-    @abstractmethod
     def name(self) -> CH_TYPES:
         """The name of the channel."""
-        pass
+        _name = type(self).__name__
+        options = get_args(CH_TYPES)
+        assert (
+            _name in options
```

Comment:
It this check to ensure that no other subclasses of `Channel` will be used? If this is the goal, it seems that it will cover many cases but not when we create a class with one of the allowed names in a different package.

Reply:
Well, it was actually more to ensure the `cast` I do afterwards is valid. I think ensuring that no other subclasses of `Channel` are used is something I could potentially do, but I would do it inside the `Device` class instead by comparing against the actually types